src/sentry/api/endpoints/user_details.py

```diff
@@ -221,15 +217,33 @@ def delete(self, request, user):
         if remaining_org_ids:
             OrganizationMember.objects.filter(
                 organization__in=remaining_org_ids,
-                user=request.user,
+                user=user,
```

Comment:
This used to just assume you were deleting yourself, this updates things so you can delete someone else (with the proper authorization of course, which happens in `UserEndpoint`)